### ISSUE TITLE
Email address related exception does not provide sufficient info on the cause of the error

### ISSUE THREAD

**Issue by ApolonTorq** [retypeapp/retype#110]
The following markdown:

```markdown
# General Info

## Fred Bloggs

**Email:** fred@bloggs.com
**Mobile:** 5555 555 555
```

Results in this error message:
```
ERROR: Specified argument was out of the range of valid values. (Parameter 'start')
System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter 'start')
          at Retype.App.Jobs.MarkdownJob.ResolveDocumentDescription(MarkdownDocument doc, String docText)
          at Retype.App.Jobs.MarkdownJob.RunAsync(MarkdownJobArgs args)
          at Retype.App.Jobs.Job`2.RunAsync(Object args, PipelineState state)
          at Retype.App.Pipelines.Pipeline`3.ExecuteAsync(TInitial initialValue, TState state)
          at Retype.App.Services.RetypeBuildService.RenderPagesAsync(Result assets, Result bonds, IReadOnlyList`1 items, IBuildOptions options, Boolean verbose)
         Please use --verbose for more details
```

There are two issues here:
- The error message gives the user very little information to go on. The following immediate questions a user has should be answered immediately in the error output without adding the --verbose switch:
  - Which markdown file, line and column did the error occur in?
  - A more user meaningful description of what the problem was.
- Should an exception occur in this scenario with the specification of an email address and @ symbol? Note that the exception doesn't occur if the email is changed to `<fred@bloggs.com>`

**Comment by fabriciomurta** [comment]
We are tracking this issue where sometimes exceptions are thrown out to the output and they shouldn't.

About the email issue,  well, that look like some kind of "edgy" edge case. :)

I could reproduce it with this minimized example:

```markdown
# a
a@a
a
```

Adding a blank line between the second and third lines avoids the problem.

**Comment by fabriciomurta** [comment]
>  Note that the exception doesn't occur if the email is changed to fred@bloggs.com

What? Isn't it already `fred@bloggs.com` in the example you provided?

Just to make it clear, this really looks like a bug. Need to look up the actual code but, I can't think how this would be "by design". I have placed it in v1.9 milestone for now but it is still pending further triage before we can determine to which version a possible fix for this is going to be included.

**Comment by ApolonTorq** [comment]
> What? Isn't it already fred@bloggs.com in the example you provided?

Modified the original post to say `<fred@bloggs.com>`. Adding the surrounding `<` and `>` stopped the exception from occuring.

**Comment by geoffreymcgill** [comment]
This email parsing issue has been fixed in Retype [`v1.11.0`](https://github.com/retypeapp/retype/releases/tag/v1.11.0). 

@ApolonTorq Thanks for reporting the issue originally! Sorry it took so long for us to get this into a published release.

Hope this helps.
